server/msteams/client.go

```diff
@@ -140,14 +141,20 @@ func NormalizeGraphAPIError(err error) error {
 				message = *terr.GetMessage()
 			}
 			return &GraphAPIError{
-				Code:    code,
-				Message: message,
+				Code:       code,
+				Message:    message,
+				StatusCode: e.ResponseStatusCode,
 			}
 		}
 	default:
+		statusCode := 0
+		if strings.HasPrefix(err.Error(), "oauth2: ") {
```

Comment:
Also, in some cases we are returining directly that the token was unable to be properly processed, I have to find a better way of doing this, but the idea is to capture oauth2 related problems to consider them a 401 error too.

Reply:
After investigating the source code, that there are some errors that are returned as `errors.New("oauth2: ...")`, so not sure if there is any better way to not use the string based approach. What I saw is that all the errors that I've seen use the `oauth2:` prefix in the library, so I think is pretty safe to use it.

Reply:
I would recommend an IsOAuthError(err) method that can handle string prefixes and error instances, and perhaps be less brittle.

Reply:
Good point, at least we encapsulate the approach in an specific point to avoid propagation of this bad pattern.